girder/api/describe.py

```diff
@@ -122,75 +130,82 @@ def responseClass(self, obj, array=False):
         self._responseClassArray = array
         return self
 
-    def param(self, name, description, paramType='query', dataType='string',
-              required=True, enum=None, default=None):
+    def _validateParamInfo(self, dataType, paramType, name):
         """
-        This helper will build a parameter declaration for you. It has the most
-        common options as defaults, so you won't have to repeat yourself as much
-        when declaring the APIs.
-
-        Note that we could expose more parameters from the Parameter Object
-        spec, for example: format, allowEmptyValue, minimum, maximum, pattern,
-        uniqueItems.
-
-        :param name: name of the parameter used in the REST query.
-        :param description: explanation of the parameter.
-        :param paramType: how is the parameter sent.  One of 'query', 'path',
-                          'body', 'header', or 'formData'.
-        :param dataType: the data type expected in the parameter. This is one
-                         of 'integer', 'long', 'float', 'double', 'string',
-                         'byte', 'binary', 'boolean', 'date', 'dateTime',
-                         'password', or 'file'.
-        :param required: True if the request will fail if this parameter is not
-                         present, False if the parameter is optional.
-        :param enum: a fixed list of possible values for the field.
+        Helper to convert and validate the dataType and paramType.
+        Prints warnings if invalid values were passed.
         """
         # Legacy data type conversions
         if dataType == 'int':
             dataType = 'integer'
-        elif dataType == 'File':
-            print(TerminalColor.warning(
-                "WARNING: dataType 'File' should be updated to 'file'"))
-            dataType = 'file'
+
+        # Parameter Object spec:
+        # If type is "file", then consumes MUST be either
```

Comment:
Can you add a mention that consumes comes from the swagger spec in that case?  The context wasn't clear to me as I read it.